test-loop-tests/src/setup/builder.rs

```diff
@@ -249,11 +229,11 @@ impl TestLoopBuilder {
             // * single shard tracking for validators
             // * all shard tracking for non-validators (RPCs and archival)
             let is_validator = genesis.config.validators.iter().any(|v| v.account_id == account_id);
-            if is_validator && !self.track_all_shards {
-                client_config.tracked_shards = Vec::new();
+            client_config.tracked_shards = if is_validator && !self.track_all_shards {
```

Comment:
Addressed this here https://github.com/near/nearcore/pull/13093#discussion_r1990090834